modules/library/render/src/main/java/org/geotools/renderer/style/SLDStyleFactory.java

```diff
@@ -1430,13 +1471,82 @@ public static int lookUpJoin(String joinType) {
 	 * @return DOCUMENT ME!
 	 */
 	public static int lookUpCap(String capType) {
-		if (SLDStyleFactory.capLookup.containsKey(capType)) {
-			return ((Integer) capLookup.get(capType)).intValue();
+		if (SLDStyleFactory.CAP_LOOKUP.containsKey(capType)) {
+			return ((Integer) CAP_LOOKUP.get(capType)).intValue();
 		} else {
 			return java.awt.BasicStroke.CAP_SQUARE;
 		}
 	}
 
+    /**
+     * Looks up the composite from the vendor options, or returns null if no composite operation has
+     * been specified in the options
+     * 
+     * @param options
+     * @return
+     */
+    public static Composite getComposite(Map<String, String> options) {
+        return getComposite(options, 1f);
+    }
+
+    /**
+     * Looks up the composite from the vendor options, or returns null if no composite operation has
+     * been specified in the options
+     * 
+     * @param options
+     * @return
+     */
+    public static Composite getComposite(Map<String, String> options, float defaultOpacity) {
+        // get the spec, if no spec, no composite
+        String spec = options.get(FeatureTypeStyle.COMPOSITE);
+        if (spec == null) {
+            return null;
+        }
+
+        // parse name or name,opacity
+        String name;
+        float opacity = defaultOpacity;
+        if (spec.indexOf(',') != -1) {
+            String[] split = spec.split("\\s*,\\s*");
+            if (split.length != 2) {
+                throw new IllegalArgumentException("Invalid syntax for "
+                        + FeatureTypeStyle.COMPOSITE
+                        + " key, expecting 'name' or 'name,opacity' but got " + spec);
+            }
+            name = split[0].trim();
+            boolean invalidOpacity = false;
+            try {
+                opacity = Float.parseFloat(split[1]);
+            } catch (NumberFormatException e) {
+                invalidOpacity = true;
+            }
+            if (invalidOpacity || opacity < 0 || opacity > 1) {
+                throw new IllegalArgumentException(
+                        "Invalid value for composite opacity, expecting a number between 0 and 1, but got '"
+                                + split[1] + "' instead");
+            }
+        } else {
+            name = spec;
+        }
+
+        // lookup the composition operation, see if it's an alpha composite
+        if (ALPHA_COMPOSITE_LOOKUP.containsKey(name)) {
+            Integer rule = ALPHA_COMPOSITE_LOOKUP.get(name);
+            return AlphaComposite.getInstance(rule, opacity);
```

Comment:
AlphaComposite has a special internal optimization for opacity = 1, no issue here, it will return a static constant (I've copied the same optimization approach in BlendComposite)